tool/src/js/gen.rs

```diff
@@ -413,7 +424,11 @@ impl<'tcx> TyGenContext<'_, 'tcx> {
             wraps_primitive: bool,
             owns_wrapped_primitive: bool,
 
-            docs: String,
+            doc_str: String,
+
+            /// Used by `js_class.js.jinja`. If a constructor isn't overridden by #[diplomat::attr(auto, constructor)], this is the logic that `js_class.js.jinja` will use to determine whether or not to generate constructor code.
+            /// Useful for hiding the fact that an out_struct has a constructor in typescript headers, for instance.
+            show_default_ctor: bool,
```

Comment:
thought (followup): we should probably move shared types into a shared ImplTemplate type, and have specific StructImplTemplate/etc